libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/util/CompositeMapboxNavigationObserver.kt

```diff
@@ -0,0 +1,23 @@
+package com.mapbox.navigation.dropin.util
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+internal class CompositeMapboxNavigationObserver(
```

Comment:
This class can be easily abused with the add/remove functions. 

``` kotlin
val myComposite = CompositeMapboxNavigationObserver()
myComposite.add(otherObserver)
MapboxNavigatationApp.register(myComposite)
MapboxNavigatationApp.attach(lifecycleOwner)
myComposite.remove(otherObserver)
```

otherObserver.onDetached(..) will never be called. All of our observers are designed to have a guaranteed call to `onDetached`.

Reply:
That's not how composite listeners work. Anyway. I am removing this class.